WIP - Wqp 1381 DO NOT MERGE

mbucknell opened the pull request:
All the ETL works. I can't seem to get the application to use my application.yml that is in resources and also can't figure out how to run the job everytime the command is run. At this point you can probably skip all of the packages under ETL (they follow a similar pattern) except for gov.acwi.wqp.etl.parm and gov.acwi.wqp.etl.methwithCit which is the one I had to do the ETL with tasklet.

I'm still looking into mocks for natdb, but did write a unit test for BaseProcessor.

mbucknell committed:
Update gitignore

mbucknell committed:
Initial pom for spring-batch application

mbucknell committed:
Initial commit of etl that etl's the altitude_method table.

mbucknell committed:
Refactored the AltitudeMethod flow into its own package.

mbucknell committed:
Refactored to create a BasicLookup types, processor and delete method so
this code can be reused for other lookups.

mbucknell committed:
Created an altitudeMethod and aquiferType directories and but the etl
code for those in there.

mbucknell committed:
Renamed BaseDeleteTable

mbucknell committed:
Updates to transformation

mbucknell committed:
Added and fixed application.yml

mbucknell committed:
Whitespace issues

mbucknell committed:
Added aqfr table etl. Also fixed more formatting issues

mbucknell committed:
More cleanup

mbucknell committed:
Added body_part etl

mbucknell committed:
Added ETL for cit_meth

mbucknell committed:
Fixed package names

mbucknell committed:
Fixed whitespace

mbucknell committed:
Created a BaseProcessor class that includes the protected trimString
method. The *Processor classes now extend this and use the trimString
method to transform string fields.

mbucknell committed:
Implemented country etl

mbucknell committed:
Implement County ETL

mbucknell committed:
Implemented Fxd ETL

mbucknell committed:
Implemented ETL for hyd_cond_cd

mbucknell committed:
Implemented hyd_event_cd ETL

mbucknell committed:
Implement lat_long_datum etl

mbucknell committed:
Implement meth ETL

mbucknell committed:
Implemented the meth_with_cit ETL. Had to use a tasklet using
JdbcTemplate rather than standard reader, processor, writer steps
because JdbcCursorItemReader failed when the query included the
aggregation function.

mbucknell committed:
Removed new String from readers

mbucknell committed:
Implemente nat_aqfr ETL

mbucknell committed:
Implemented parm_alias ETL

mbucknell committed:
Implemented most of parm ETL. still need to add adding the multiplier
column

mbucknell committed:
Additional tasklet to update the parm table's meth column

mbucknell committed:
Implemented proto_org ETL

mbucknell committed:
Implemented site_tp etl

mbucknell committed:
Implemented state ETL

mbucknell committed:
Implemented tu ETL

mbucknell committed:
Implement val_qual_cd ETL

mbucknell committed:
Added a unit test for BaseProcessor. Some cleanup and cleanup of
application.yml resources

dsteinich commented on pom.xml:
```diff
@@ -0,0 +1,102 @@
+    </parent>
+
+    <properties>
+        <java.version>1.8</java.version>
```
Boot has a default java.version of 1.8. We should probably get in the habit of setting it to 11.

dsteinich commented on pom.xml:
```diff
@@ -0,0 +1,102 @@
+            <groupId>com.oracle</groupId>
+            <artifactId>ojdbc6</artifactId>
+            <version>11.2.0.2.0-1</version>
+            <scope>provided</scope>
```
Maybe take the scope off as we will need it included in the jar (will not be deployed to a tomcat containing it...)

dsteinich commented on src/main/java/gov/acwi/wqp/etl/BaseDeleteTable.java:
```diff
@@ -0,0 +1,29 @@
+
+	@Override
+	public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
+		String sql = new String("truncate table " + tableName);
```
hmmm, still need that sql injection proofing?

mbucknell committed:
Initial README, added BasicLookupProcessor test and using a prepared
statement in TransformBasicLookup

dsteinich commented on src/main/java/gov/acwi/wqp/etl/BasicLookupProcessor.java:
```diff
@@ -0,0 +1,17 @@
+		final String description = trimString(gwReflist.getGwRefDs());
+		final boolean validFlag = gwReflist.getGwValidFg() == 'Y';
+		
+		return new BasicLookup(code, name, gwReflist.getGwSortNu(), description, validFlag);
```
Mixed thoughts on this - the "final" seems like noise (but if I knew JAVA, I would know why it is or shouldn't be here...)
- not sure on some of the contructor parms being variables and one a method (I know it isn't wrapped in a function like the others)
- 5 is probably ok, but much more (coupled with the data manipulation and fact that we are in a processor) and I may have not done the constructor for BasicLookup.

mbucknell replied:
What would be the alternative? I use this pattern in all of the processors, some of which have a lot more parameters.

mbucknell replied:
I think you are correct that "final" is noice and I'm fine with making all of the constructor parms variables. Still not sure what you mean by the second bullet point.

mbucknell replied:
Now that I reread the last bullet, are you suggesting that it would be better to use the setters?

mbucknell committed:
Updated to have the job always run when the jar is invoked.

mbucknell committed:
Changes to pom requested by review.

mbucknell committed:
Using function to truncate tables

mbucknell committed:
Uses setters rather than constructor.

mbucknell commented on src/main/java/gov/acwi/wqp/etl/BasicLookupProcessor.java:
```diff
@@ -0,0 +1,19 @@
+		result.setDescription(trimString(gwReflist.getGwRefDs()));
+		result.setValidFlag(gwReflist.getGwValidFg() == 'Y');
+		
+		return result;
```
@dsteinich is this what you are recommending?

mbucknell committed:
Removed the TransformBasicLookup class and implemented the four basic
lookup ETL transformations directly. This prevents the appearance of
using SQL injection

mbucknell committed:
Updated the *Processor classes to use setters rather than constructors.

mbucknell committed:
Moved the variable declarations in the class before the method
declarations

mbucknell committed:
Fixed more Codacy flagged issues

mbucknell committed:
More codacy linting issue fixes

mbucknell committed:
One more linting issue

mbucknell committed:
One more linting issue

mbucknell commented:
Will close this and open one for actually merging the code. I have not come up with an effective way of mocking natdb.
